src/praisonai-agents/praisonaiagents/llm/llm.py

```
@@ -923,6 +923,97 @@ def _refresh_subscription_creds(self):
+        if category == "auth" and self._auth_provider_id and attempt == 0:
+            try:
+                logging.info("Authentication error detected - attempting credential refresh")
+                refreshed_creds = self._refresh_subscription_creds()
+                if refreshed_creds:
+                    # Update parameters with refreshed credentials (don't clear cache)
+                    kwargs = self._build_completion_params(**kwargs)
+                    # Retry immediately with refreshed credentials
+                    can_retry = True
+                    retry_delay = 0.0
+                    logging.info("Subscription credentials refreshed, retrying...")
+            except Exception as refresh_error:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)